src/elysium/txprocessor.cpp

```diff
@@ -154,18 +155,53 @@ int TxProcessor::ProcessSimpleSpend(const CMPTransaction& tx)
     }
 
     auto spend = tx.getSpend();
+    auto serial = tx.getSerial();
     auto denomination = tx.getDenomination();
     auto group = tx.getGroup();
     auto groupSize = tx.getGroupSize();
 
     bool const fPadding = block >= ::Params().GetConsensus().nSigmaPaddingBlock;
 
     assert(spend);
+    assert(serial);
+
+    if (IsFeatureActivated(FEATURE_SIGMA_SPENDV1, block)) {
+        if (!serial->isMember()) {
```

Comment:
We need to add `|| serial->isZero()`  here

Reply:
@levonpetrosyan93 After discussing with @ultimaweapon we should remove this check because we should let the protocol layer deal with this instead of rejecting in consensus layer. How do you think?

Reply:
To be more clearer we would let function `VerifySigmaSpend` handle it.

Reply:
but looks  have no such checks at protocol layer,

Reply:
Why we do not make it have that check at the cryptographic level?

Reply:
@ultimaweapon if you mean lib sigma, the interface doesn't accept serial we use it before passing to the library(function `Verify` in file src/elysium/sigmaprimitives.h). Anyway, after research and discussion with @levonpetrosyan93 we agree to delete the check here.

Reply:
At Sigma we have check for serial at CoinSpend when it is checked to be the hash of pubkey, I think it is enough, but at elysium code I could not find such check.

Reply:
In Elysium we generate serial from pubkey here https://github.com/thebevrishot/zcoin/blob/rename-exodus/src/elysium/tx.cpp#L875